crates/cubecl-cuda/src/compiler/instruction.rs

```diff
@@ -463,3 +465,41 @@ impl Remainder {
         Ok(())
     }
 }
+
+macro_rules! unique_variable_name {
+    ($base:expr) => {
+        concat!("__", $base, "_", line!())
+    };
+}
+
+struct Normalize;
+
+impl Normalize {
+    fn format(
+        f: &mut core::fmt::Formatter<'_>,
+        input: &Variable,
+        out: &Variable,
+    ) -> core::fmt::Result {
+        let num = input.item().vectorization;
+        let norm = unique_variable_name!("norm");
+
+        f.write_fmt(format_args!("{{\n"))?;
+        f.write_fmt(format_args!("float {norm} = 0.0;\n"))?;
+
+        for i in 0..num {
+            let input_i = input.index(i);
+            f.write_fmt(format_args!("{norm} += {input_i} * {input_i};\n"))?;
+        }
+
+        f.write_fmt(format_args!("{norm} = sqrt({norm});\n"))?;
```

Comment:
`sqrt` might have a different name based on the element type. See how it is implemented in `unary.rs`

Maybe we could create a new variable 

```rust
enum Varialbe {
   Named(String, Item),
}
```

And reuse the other instructions to build the algo. Just a though.

Reply:
I added a helper function for the function prefixes, that could easily be reused in unary.rs as well, I'm just not sure where to put it.

Reply:
I changed the helper function, it worked but it felt a bit weird. Instead you can now write ```rust
Sqrt::format_unary(f, input, out, elem)?;
```
and it works for any input and output that are ```Display```. It also automatically works for any `Unary`, not just sqrt. I made an additional `FunctionFmt` trait that can add the function name prefixes for you, I had to do this since `Unary` has the downside of requiring `Component` in it's formatting, which is necessary for `Assign`.

An alternative way of doing it could be by integrating `FunctionFmt` straight into Unary, but then that approach wouldn't work for `Assign` again.

I possibly see what you mean now with the `Variable` suggestion, that would make this work with `Assign` and other functions, by not needing to use `String` anymore for temporary variables.